src/test/java/com/google/research/bleth/simulator/TracingSimulationIT.java

```diff
@@ -13,65 +16,159 @@
-        Location beaconExpectedLocation = predictLocationAfterMoveUp(beaconInitialLocation);
-        Location observerExpectedLocation = observerInitialLocation;
+        Map<IAgent, Location> initialAgentsToLocations = mapAgentsToLocationsOnBoard(simulation.getBoard());
 
-        // Run single-rounded simulation and find agents again.
         simulation.run();
-        Location beaconActualLocation = findAgent(simulation.getBoard(), "Beacon");
-        Location observerActualLocation = findAgent(simulation.getBoard(), "Observer");
+        Map<IAgent, Location> terminalAgentsToLocations = mapAgentsToLocationsOnBoard(simulation.getBoard());
 
-        assertThat(beaconActualLocation).isEqualTo(beaconExpectedLocation);
-        assertThat(observerActualLocation).isEqualTo(observerExpectedLocation);
+        for (IAgent agent : initialAgentsToLocations.keySet()) {
+            assertThat(terminalAgentsToLocations.get(agent)).isEqualTo(initialAgentsToLocations.get(agent));
+        }
     }
 
-    private Location findAgent(Board board, String type) {
-        Multimap<Location, IAgent> agentsOnLocations = board.agentsOnBoard();
-        for(int row = 0; row < board.getRowNum(); row++) {
-            for(int col = 0; col < board.getColNum(); col++) {
-                Location loc = Location.create(row, col);
-                for(IAgent agent : agentsOnLocations.get(loc)) {
-                    if (agent.getId() == 0 && agent.getType().equals(type)) {
-                        return loc;
-                    }
-                }
-            }
+    @Test
+    public void runTwoRoundsSimulationVerifyBeaconLocationOnEstimatedBoardAccordingToObserverLocation() {
+        int roundsNum = 2;
+        int rowsNum = 2;
+        int colsNum = 2;
+        int beaconsNum = 1;
+        int observersNum = 1;
+        double transmissionRadius = 2.0;
+
+        AbstractSimulation simulation = new TracingSimulation.Builder()
+                .setMaxNumberOfRounds(roundsNum + 1) // The first round is the initialization
+                .setRowNum(rowsNum)
+                .setColNum(colsNum)
+                .setBeaconsNum(beaconsNum)
+                .setObserversNum(observersNum)
+                .setTransmissionThresholdRadius(transmissionRadius)
+                .setBeaconMovementStrategy(MOVE_UP)
+                .setObserverMovementStrategy(STATIONARY)
+                .setAwakenessCycle(AWAKENESS_CYCLE_EQUALS_TWO)
+                .setAwakenessDuration(AWAKENESS_DURATION_EQUALS_ONE)
+                .setAwakenessStrategyType(FIXED_AWAKENESS_STRATEGY_TYPE)
+                .build();
+
+        Map<IAgent, Location> initialAgentsToLocations = mapAgentsToLocationsOnBoard(simulation.getBoard());
+        Set<IAgent> observers = initialAgentsToLocations.keySet().stream().filter(agent -> agent instanceof Observer).collect(Collectors.toSet());
+        Location observerInitialLocation = averageLocationOfObservers(observers);
```

Comment:
We have a single observer here, why do we compute the average location?
(If the reason is to fit the pattern of the other tests, that's fine, just wanted to make sure).